tests/ops/op_math/test_sprod.py

```diff
@@ -0,0 +1,635 @@
+    (1.0 + 2.0j, qml.Identity(wires=0)),
+    (10, qml.IsingXX(4.56, wires=[2, 3])),
+    (0j, qml.Toffoli(wires=[1, 2, 3])),
+    (42, qml.Rot(0.34, 1.0, 0, wires=0)),
+)
+
+ops_rep = (
+    "1.0*(PauliX(wires=[0]))",
+    "0.0*(PauliZ(wires=[0]))",
+    "1j*(Hadamard(wires=[0]))",
+    "1.23*(CNOT(wires=[0, 1]))",
+    "4.56*(RX(1.23, wires=[1]))",
+    "(1+2j)*(Identity(wires=[0]))",
+    "10*(IsingXX(4.56, wires=[2, 3]))",
+    "0j*(Toffoli(wires=[1, 2, 3]))",
+    "42*(Rot(0.34, 1.0, 0, wires=[0]))",
+)
+
+
+def get_qft_mat(num_wires):
+    """Helper function to generate the matrix of a qft protocol."""
+    omega = math.exp(np.pi * 1.0j / 2 ** (num_wires - 1))
+    mat = math.zeros((2**num_wires, 2**num_wires), dtype="complex128")
+
+    for m in range(2**num_wires):
+        for n in range(2**num_wires):
+            mat[m, n] = omega ** (m * n)
+
+    return 1 / math.sqrt(2**num_wires) * mat
+
+
+class TestInitialization:
+    """Test initialization of ther SProd Class."""
+
+    @pytest.mark.parametrize("test_id", ("foo", "bar"))
+    def test_init_sprod_op(self, test_id):
+        sprod_op = s_prod(3.14, qml.RX(0.23, wires="a"), do_queue=True, id=test_id)
+
+        assert (
+            sprod_op.scalar == 3.14
+        )  # no need to test if op.base == RX since this is covered in SymbolicOp tests
+        assert sprod_op.wires == Wires(("a",))
+        assert sprod_op.num_wires == 1
+        assert sprod_op.name == "SProd"
+        assert sprod_op.id == test_id
+
+        assert sprod_op.data == [[3.14], [0.23]]
+        assert sprod_op.parameters == [[3.14], [0.23]]
+        assert sprod_op.num_params == 2
+
+    def test_queue_idx(self):
+        sprod_op = s_prod(3.14, qml.Identity(wires=1))
+        assert sprod_op.queue_idx is None
+
+    def test_parameters(self):
+        sprod_op = s_prod(9.87, qml.Rot(1.23, 4.0, 5.67, wires=1))
+        assert sprod_op.parameters == [[9.87], [1.23, 4.0, 5.67]]
+
+    def test_data(self):
+        sprod_op = s_prod(9.87, qml.Rot(1.23, 4.0, 5.67, wires=1))
+        assert sprod_op.data == [[9.87], [1.23, 4.0, 5.67]]
+
+    def test_data_setter(self):
+        """Test the setter method for"""
+        scalar, angles = (9.87, (1.23, 4.0, 5.67))
+        old_data = [[9.87], [1.23, 4.0, 5.67]]
+
+        sprod_op = s_prod(scalar, qml.Rot(*angles, wires=1))
+        assert sprod_op.data == old_data
+
+        new_data = [[1.23], [0.0, -1.0, -2.0]]
+        sprod_op.data = new_data
+        assert sprod_op.data == new_data
+        assert sprod_op.scalar == new_data[0][0]
+        assert sprod_op.base.data == new_data[1]
+
+    @pytest.mark.parametrize("scalar, op", ops)
+    def test_terms(self, op, scalar):
+        sprod_op = SProd(scalar, op)
+        coeff, ops = sprod_op.terms()
+
+        assert coeff == [scalar]
+        for op1, op2 in zip(ops, [op]):
+            assert op1.name == op2.name
+            assert op1.wires == op2.wires
+            assert op1.data == op2.data
```

Comment:
We have a `qml.equal` now!
```suggestion
            assert qml.equal(op1, op2)
```